src/coreclr/jit/llvm.cpp

```diff
@@ -896,9 +1106,11 @@ N011 ( 34, 17) [000016] DA-XG-------              *  STORE_LCL_VAR int    V03 lo
 */
 Value* widenIntIfNecessary(llvm::IRBuilder<>& builder, Value* intValue)
 {
-    if (intValue->getType()->getPrimitiveSizeInBits() < TARGET_POINTER_SIZE * 8)
+    llvm::TypeSize intSize = intValue->getType()->getPrimitiveSizeInBits();
+    if (intSize < TARGET_POINTER_SIZE * 8)
     {
-        return builder.CreateIntCast(intValue, Type::getInt32Ty(_llvmContext), true);
+        bool isSigned = intSize > 1; // bools are not signed;
```

Comment:
What about other small unsigned types?

Reply:
Yes, this was broken I think but the methods in question were failing on other missing functionality.  I changed this to always be a zero extend and changed the method name.  There is some clean up to be done here, but I've just remarked on it and added a TODO.